e2e_tests/tests/conftest.py

```diff
@@ -206,3 +208,64 @@ def test_start_timer(request: _pytest.fixtures.SubRequest) -> Iterator[None]:
         # This ends up concatenated to the line pytest prints containing the test file and name.
         print("starting at", time.strftime("%Y-%m-%d %H:%M:%S"))
     yield
+
+
+@pytest.fixture(scope="session")
+def is_multirm_cluster() -> bool:
+    sess = api_utils.admin_session()
+    multi_rm = True
+    try:
+        multi_rm = detproc.check_json(sess, ["det", "master", "config", "show", "--json"])[
+            "additional_resource_managers"
+        ]
+    except KeyError:
+        multi_rm = False
+    return multi_rm
+
+
+@pytest.fixture(scope="session")
+def namespaces_created(is_multirm_cluster: bool) -> Tuple[str, str]:
+    defaultrm_namespace = uuid.uuid4().hex[:8]
+    additionalrm_namespace = uuid.uuid4().hex[:8]
+
+    # Create a namespace in Kubernetes in the each resource manager's Kubernetes cluster.
+    create_namespace_defaultrm_cmd = ["kubectl", "create", "namespace", defaultrm_namespace]
+
+    if is_multirm_cluster:
+        create_namespace_defaultrm_cmd += ["--kubeconfig", conf.DEFAULT_RM_KUBECONFIG]
+        create_namespace_additionalrm_cmd = [
+            "kubectl",
+            "create",
+            "namespace",
+            additionalrm_namespace,
+            "--kubeconfig",
+            conf.ADDITIONAL_RM_KUBECONFIG,
+        ]
+        subprocess.run(create_namespace_additionalrm_cmd, check=True)
+
+    subprocess.run(create_namespace_defaultrm_cmd, check=True)
+
+    default_kubeconfig = []
+    if is_multirm_cluster:
+        get_namespace(additionalrm_namespace, ["--kubeconfig", conf.ADDITIONAL_RM_KUBECONFIG])
+        default_kubeconfig += ["--kubeconfig", conf.DEFAULT_RM_KUBECONFIG]
+
+    get_namespace(defaultrm_namespace, default_kubeconfig)
+
+    # Make sure that we can successfully retrieve both namespaces.
+    namespaces = [defaultrm_namespace]
+    if is_multirm_cluster:
+        namespaces.append(additionalrm_namespace)
+
+    return defaultrm_namespace, additionalrm_namespace
+
+
+def get_namespace(namespace: str, kubeconfig: List[str]) -> None:
+    for _ in range(150):
+        try:
+            p = subprocess.run(["kubectl", "get", "namespace", namespace] + kubeconfig, check=True)
+            if not p.returncode:
+                break
```

Comment:
Shouldn't this just be "break", since the `check=True` will have already raised an exception if returncode is not zero, aborting evaluation of the try block? The `if not returncode` seems to reduce readability, IMO (it takes a some thinking to "oh, yeah, 0 is false but exiting true").

Reply:
Perhaps even easier, kubectl has wait.

`kubectl wait --for jsonpath='{.status.phase}=Active' --timeout=300s namespace/default`

Reply:
Ref [above reply](https://github.com/determined-ai/determined/pull/9180#discussion_r1690073977)